src/CodeGen/ODataT4CodeGenerator.ttinclude

```diff
@@ -317,6 +326,45 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+public void ValidateAndSetCustomHttpHeadersFromString(string stringValue){
+    var defaultReturn = new Dictionary<string, string>();
+    if(String.IsNullOrWhiteSpace(stringValue)){
+        this.CustomHttpHeaders = defaultReturn;
+        return;
+    }
+        
+    stringValue = stringValue
+        // remove surrounding "[]"
+        .Replace("[", "").Replace("]", "");
+
+    if(String.IsNullOrWhiteSpace(stringValue)){
+        this.CustomHttpHeaders = defaultReturn;
+        return;
+    }
+    if(
+        // Check if we have any valid object in the string
+        !stringValue.Contains("{") || !stringValue.Contains("}") || 
+        // Check if the count of the keys are the same as the count of values
+        (stringValue.Count(d => d == ':') != stringValue.Count(d => d == '{')) ||
+        // Check all objects of a '{' and a '}'
+        (stringValue.Count(d => d == '{') != stringValue.Count(d => d == '}')) ||
+        // Check all values and keys are strings
+        (stringValue.Count(d => d == ':')*4 != stringValue.Count(d => d == '\''))
+        ){
+        throw new ArgumentException("Please check your custom http headers string format!");
+    }

```

Comment:
I don't know that these validations are valid; for example, a colon is a valid (and often used) character within a key-value.